core/modules/replica/Configuration.cc

```diff
@@ -39,79 +39,163 @@ namespace {
 
 LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.Configuration");
 
+/**
+ * @param connectionUrl The connection URL.
+ * @param database The optional name of a database to replace the one defined in the url.
+ * @return The MySQL connection descriptor.
+ */
+database::mysql::ConnectionParams connectionParams(string const& connectionUrl, string const& database) {
+    database::mysql::ConnectionParams params = database::mysql::ConnectionParams::parse(connectionUrl);
+    if (!database.empty()) params.database = database;
+    return params;
+}
 } // namespace
 
 namespace lsst {
 namespace qserv {
 namespace replica {
 
+
 // These (static) data members are allowed to be changed, and they are set
 // globally for an application (process).
 bool         Configuration::_databaseAllowReconnect = true;
 unsigned int Configuration::_databaseConnectTimeoutSec = 3600;
 unsigned int Configuration::_databaseMaxReconnects = 1;
 unsigned int Configuration::_databaseTransactionTimeoutSec = 3600;
-string       Configuration::_qservMasterDatabasePassword = "";
-string       Configuration::_qservWorkerDatabasePassword = "";
+string       Configuration::_qservCzarDbUrl = "mysql://qsreplica@localhost:3306/qservMeta";
+string       Configuration::_qservWorkerDbUrl = "mysql://qsreplica@localhost:3306/qservw_worker";
 bool         Configuration::_xrootdAllowReconnect = true;
 unsigned int Configuration::_xrootdConnectTimeoutSec = 3600;
+util::Mutex  Configuration::_classMtx;
 
 
-void Configuration::setQservMasterDatabasePassword(string const& newPassword) {
-    string result = newPassword;
-    swap(result, _qservMasterDatabasePassword);
+// ---------------
+// The static API.
+// ---------------
+
+void Configuration::setQservCzarDbUrl(string const& url) {
+    util::Lock const lock(_classMtx, _context(__func__));
+    if (url.empty()) {
```

Comment:
This check can be done before locking, not that it matters much in this case but generally it's a good practice.

Reply:
Good point.